packages/core/src/utils/helpers.ts

```diff
@@ -28,3 +28,26 @@ export function notNull<T>(value: T | null | undefined): value is T {
 }
 
 export function noop() {}
+
+export interface Colors {
+  [key: string]: Colors | string
+}
+
+export function assignDefaultToColors(colors: Colors): void {
+  Object.values(colors).forEach((color) => {
+    if (typeof color !== 'string' && color !== undefined) 
+      color.DEFAULT = color.DEFAULT || color[400]
```

Comment:
I wonder if we could go recursive to support deep colors

```ts
const colors = {
    red: {
      50: '#fef2f2',
      100: '#fee2e2',
      200: '#fecaca',
      deep: {
        100: '#...',
        DEFAULT: '#...'
      }
    }
  }
```

We didn't do that because it was only for the built-in colors object

Reply:
IMO we could do this in a separate PR

Reply:
Then I don't actually see the value of exposing those utilities. This code was not meant to be used as utilities, they are just to keep the bundle small and make the codebase more maintainable (as all are static). Do you want to use them externally to your own theme object?